hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -674,6 +685,804 @@ public Response getDeletedDirectorySummary() {
+            keyIter.seek(seekKey);
+
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of an empty prevKeyPrefix, all the keys are returned
+        if (seekKeyValue == null ||
+            (StringUtils.isNotBlank(paramInfo.getPrevKey()) &&
+                !seekKeyValue.getKey().equals(paramInfo.getPrevKey()))) {
+          return matchedKeys;
+        }
+      } else {
+        keyIter.seek(paramInfo.getStartPrefix());
+      }
+
+      while (keyIter.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> entry = keyIter.next();
+        String dbKey = entry.getKey();
+        paramInfo.setLastKey(dbKey);
+        if (!dbKey.startsWith(paramInfo.getStartPrefix())) {
+          break; // Exit the loop if the key no longer matches the prefix
+        }
+        if (skipPrevKey && dbKey.equals(paramInfo.getPrevKey())) {
+          paramInfo.setSkipPrevKeyDone(true);
+          continue;
+        }
+        if (applyFilters(entry, paramInfo) && matchedKeys.size() < paramInfo.getLimit()) {
+          matchedKeys.put(dbKey, entry.getValue());
+        }
+      }
+    } catch (IOException exception) {
+      LOG.error("Error retrieving keys from table for path: {}", paramInfo.getStartPrefix(), exception);
+      throw exception;
+    }
+    return matchedKeys;
+  }
+
+  private boolean applyFilters(Table.KeyValue<String, OmKeyInfo> entry, ParamInfo paramInfo) throws IOException {
+
+    LOG.debug("Applying filters on : {}", entry.getKey());
+
+    long epochMillis =
+        ReconUtils.convertToEpochMillis(paramInfo.getCreationDate(), "MM-dd-yyyy HH:mm:ss", TimeZone.getDefault());
+    Predicate<Table.KeyValue<String, OmKeyInfo>> keyAgeFilter = keyData -> {
+      try {
+        return keyData.getValue().getCreationTime() >= epochMillis;
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    };
+    Predicate<Table.KeyValue<String, OmKeyInfo>> keyReplicationFilter =
+        keyData -> {
+          try {
+            return keyData.getValue().getReplicationConfig().getReplicationType().name()
+                .equals(paramInfo.getReplicationType());
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
+        };
+    Predicate<Table.KeyValue<String, OmKeyInfo>> keySizeFilter = keyData -> {
+      try {
+        return keyData.getValue().getDataSize() > paramInfo.getKeySize();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    };
+
+    List<Table.KeyValue<String, OmKeyInfo>> filteredKeyList = Stream.of(entry)
+        .filter(keyData -> !StringUtils.isEmpty(paramInfo.getCreationDate()) ? keyAgeFilter.test(keyData) : true)
+        .filter(
+            keyData -> !StringUtils.isEmpty(paramInfo.getReplicationType()) ? keyReplicationFilter.test(keyData) : true)
+        .filter(keySizeFilter)
+        .collect(Collectors.toList());
```

Comment:
The `anyMatch` method can beused instead of collecting the filtered entries to a list and checking its size. This makes the code more efficient as the stream will short-circuit on the first match.

Reply:
As discussed, anyMatch just matches any one element in stream and terminate. We need apply filters on all elements. So this may not suit our usecase.